presto-main/src/main/java/com/facebook/presto/util/DateTimeUtils.java

```diff
@@ -404,6 +403,7 @@ private static PeriodFormatter cretePeriodFormatter(IntervalField startField, In
                     break;
                 }
                 builder.appendLiteral("-");
+                // fall through
```

Comment:
Is this require for checkstyle to pass? That seems lame :(

Reply:
If we assume accidental fall through never happens, then we do not need checkstyle check (nor IntelliJ inspection).
Otherwise, how would the check know this fall through is intended?

Reply:
I agree with @findepi. I don't find this lame. Both @dain and I are already doing `// fall through` in other places in the code before this PR. On the other hand, I would call `//CHECKSTYLE.OFF` and `//CHECKSTYLE.ON` lame.

Reply:
Both are lame. You're effectively adding extra-linguistic syntax to affect the compilation-time semantics of code involving `switch`. Comments as syntax suffer from long-term decay problems: IDEs are unaware of them, people may not know they have semantic meaning and may move them around, logic may change such that the code no longer falls through, yet the comment remains.

Reply:
I would even argue that because there are legitimate uses for fall-through switch statements, we should remove that checkstyle validation altogether.

Reply:
@martint 

I disagree. I believe our disagreement comes from this: I consider it a bad design (or bug in language specification) when a language does fall-through by default. In my opinion<sup>1</sup>, there are only two valid options for a language:

* no fall through as the default and only option
* require explicit `break` or `fall-through`, and has no default behavior

Java chose something out of these two, which nowadays is generally considered a terrible idea.

Fortunately, this is much better than most language specification regret in that this can be fixed in a way that is forward-compatible (but still not backward compatible) by introducing the `// fall through` syntax (admittedly not elegant). This check-style requirement is basically implementing this said bug fix to the language specification.

Footnote 1: This is probably widely accepted, not just my opinion. But since I'm not giving a citation here, I'll just say "in my opinion".